lib/dry/transaction/instance_methods.rb

```diff
@@ -80,12 +80,11 @@ def method_missing(name, *args, &block)
 
         operation.(*args, &block)
       end
-      # if step.internal? && operations[step.name]
-      #   operations[step.name]
-      # elsif methods.include?(step.name) || private_methods.include?(step.name)
 
       def resolve_operation(step, **operations)
-        if methods.include?(step.name) || private_methods.include?(step.name)
+        if step.internal? && operations[step.name]
```

Comment:
I see what you did ;) 

Is actually much simpler than the solution I proposed using polymorphism.

Just one style annotation, could we have all the good path branches first and the error ones at the end?

```ruby
if step.internal? && operations[step.name]
  operations[step.name]
elsif methods.include?(step.name) || private_methods.include?(step.name)
  method(step.name)
elsif operations[step.name].respond_to?(:call)
  operations[step.name]
elsif operations[step.name].nil?
  raise MissingStepError.new(step.name)       
else
  raise InvalidStepError.new(step.name)
end
```

Reply:
@GustavoCaso Nice idea. Pushed that.